lib/puppet_x/puppetlabs/powershell/powershell_manager.rb

```diff
@@ -41,6 +50,19 @@ def initialize(cmd)
         at_exit { exit }
       end
 
+      def alive?
+        # powershell process running
+        @ps_process.alive? &&
+          # explicitly set during a read / write failure, like broken pipe EPIPE
+          @usable &&
+          # an explicit failure state might not have been hit, but IO may be closed
+          # calling stat can yield an EBADF when underlying OS handle is bad
+          # TODO: what error is yielded when closed? is called on a bad pipe??
+          (!@stdin.closed? && !@stdin.stat.nil? rescue false) &&
```

Comment:
I think instead of calling `.stat` here, I should see if a call to `get_osfhandle` is sufficient (my hunch is that it is)